examples/plotting/time_distance.py

```diff
@@ -0,0 +1,95 @@
+"""
+=====================================================
+Creating a time-distance plot from a sequence of maps
+=====================================================
+
+This example showcases how you can use :func:`sunpy.map.pixelate_coord_path` and
+:func:`sunpy.map.sample_at_coords` on a sequence of images to create a time-distance diagram.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import propagate_with_solar_surface
+from sunpy.coordinates.screens import SphericalScreen
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+###############################################################################
+# First, we will need to acquire a sequence of images from SDO/AIA.
+
+query = Fido.search(
+    a.Time('2025-05-01 00:00:00', '2025-05-01 00:10:00'),
+    a.Instrument.aia,
+    a.Wavelength(171*u.angstrom),
+    a.Sample(1*u.min),
+)
+files = Fido.fetch(query, site="NSO")
+aia_sequence = sunpy.map.Map(files, sequence=True)
+
+###############################################################################
+# Our target will be a set of loops in the corona over a small bright region.
+# We will create a rectangular slice (i.e., a map cutout) around this region.
+
+corner = SkyCoord(Tx=800*u.arcsec, Ty=200*u.arcsec, frame=aia_sequence[0].coordinate_frame)
+cut_sequence = []
+for aia_map in aia_sequence:
+    with SphericalScreen(aia_map.observer_coordinate):
+        cut_sequence.append(aia_map.submap(corner, width=400*u.arcsec, height=400*u.arcsec))
+cut_aia_sequence = sunpy.map.Map(cut_sequence, sequence=True)
+
+###############################################################################
+# Now that we have our cutouts, we can reproject each map in our sequence to
+# the WCS of that cutout using the `~sunpy.coordinates.propagate_with_solar_surface`
+# context manager to adjust the field of view of the cutout with the rotation of the solar surface.
+
+with propagate_with_solar_surface():
+    with SphericalScreen(cut_aia_sequence[0].observer_coordinate):
+        aia_sequence_aligned = sunpy.map.Map([m.reproject_to(cut_aia_sequence[0].wcs) for m in cut_aia_sequence], sequence=True)
+
+aia_sequence_aligned.plot()
```

Comment:
Few things I'd update to use similar coordinates as in the issue so will actually see something.

I think these two block can be combined into one e.g.

```
reprojected_sub_maps = []
for cur_map in aia_seq:
    cur_map = cur_map/cur_map.exposure_time
    with propagate_with_solar_surface(), SphericalScreen(cur_map.observer_coordinate, only_off_disk=True):
        reprojected_sub_maps.append(cur_map.reproject_to(ref_sub_map.wcs))
sub_map_seq = Map(reprojected_sub_maps, sequence=True)
```

Might be worth noting the the order of the context mangers is important AFAIK